crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -3107,9 +3107,10 @@ pub(crate) fn report_invalid_key_on_typed_dict<'db>(
             }
             _ => {
                 let mut diagnostic = builder.into_diagnostic(format_args!(
-                    "Invalid key of type `{}` for TypedDict `{}`",
-                    key_ty.display(db),
+                    "TypedDict `{}` can only be subscripted with string literal keys, \
```

Comment:
for consistency with the other message, which now uses the phrase "a string literal key" rather than "string literal keys"

```suggestion
                    "TypedDict `{}` can only be subscripted with a string literal key, \
```